iceoryx_hoofs/source/file_reader/file_reader.cpp

```diff
@@ -33,34 +35,31 @@ FileReader::FileReader(const std::string& f_fileName, const std::string& f_fileP
 
     if (!isOpen())
     {
-        errorHandler(Error::kFILEREADER__FAILED_TO_OPEN_FILE, [=]() {
-            switch (m_errorMode)
-            {
-            case ErrorMode::Ignore:
-            {
-                break;
-            }
-            default:
-            case ErrorMode::Inform:
-            {
-                std::cerr << "\033[5;31m"
-                          << "Could not open file '" << m_file << "'."
-                          << "\033[0m" << std::endl;
-                break;
-            }
-            case ErrorMode::Terminate:
-            {
-                std::cerr << "\033[5;31m"
-                          << "Could not open file '" << m_file << "'. Exiting!"
-                          << "\033[0m" << std::endl;
-                std::terminate();
-                break;
-            }
-            }
-        });
+        switch (m_errorMode)
+        {
+        case ErrorMode::Ignore:
+        {
+            return;
+        }
+        default:
+        case ErrorMode::Inform:
+        {
+            std::cerr << "\033[5;31m"
+                      << "Could not open file '" << m_file << "'."
+                      << "\033[0m" << std::endl;
+            return;
+        }
+        case ErrorMode::Terminate:
+        {
+            std::cerr << "\033[5;31m"
+                      << "Could not open file '" << m_file << "'. Exiting!"
+                      << "\033[0m" << std::endl;
+            cxx::Ensures(false);
```

Comment:
Since we will most likely use the error handler in the complete code base, why not use it here with a newly created enum to spare us some future work.

Reply:
I disagree with using the `errorHandler()` directly in `iceoryx_hoofs`. See [this comment](https://github.com/eclipse-iceoryx/iceoryx/pull/1100#discussion_r807013547). That's something for #1032. For `v2.0` I would like to implement our current error handling strategy as described in `doc/design/error-handling.md`.